### ISSUE TITLE
Current version of StyleCop  Analyzers cannot handle collection expressions

### ISSUE THREAD

**Issue by DavidArno** [DotNetAnalyzers/StyleCopAnalyzers#3706]
One feature likely coming to C# 12 next month is [collection expressions](https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-12#collection-expressions). 

An example of such an expression is:
```cs
int[][] twoD = [[1, 2, 3], [4, 5, 6], [7, 8, 9]];
```

StyleCopAnalyzers version 1.2.0-beta.507 complains three times about the formatting of this expression:

> `SA1010 - Opening square brackets should not be preceded by a space`

Remove the spaces though so it's written as:
```cs
int[][] twoD =[[1, 2, 3],[4, 5, 6],[7, 8, 9]];
```

And the analyzers now complain that:

> `SA1001 - Commas should be followed by whitespace`

Rule SA1010 needs updating so it does not report on collection-expressions.

**Comment by bjornhellander** [comment]
Looks very much like a duplicate of #3687
